### ISSUE TITLE
False positive with `svelte/no-navigation-without-resolve` when `href` is `null` or `undefined`

### ISSUE THREAD

**Issue by Stadly** [sveltejs/eslint-plugin-svelte#1336]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [x] I have tried restarting my IDE and the issue persists.
- [x] I have updated to the latest version of the packages.

### What version of ESLint are you using?

9.34.0

### What version of `eslint-plugin-svelte` are you using?

3.12.2

### What did you do?

Updated to eslint-plugin-svelte v3.12

```svelte
<a href={undefined}>test</a>
<a href={null}>test</a>
```

### What did you expect to happen?

`href` can be `null` or `undefined`:
<img width="766" height="260" alt="Image" src="https://github.com/user-attachments/assets/71426c17-603e-4a31-a8f5-03ecec5e1a70" />

`svelte/no-navigation-without-resolve` should not require `null` or `undefined` to be `resolve`d.

### What actually happened?

`error  Found a link with a url that isn't resolved  svelte/no-navigation-without-resolve`

### Link to **GitHub Repo** with Minimal Reproducible Example

I'll provide one if needed.
